examples/flutter_gallery/lib/demo/text_field_demo.dart

```diff
@@ -79,26 +112,19 @@ class TextFieldDemoState extends State<TextFieldDemo> {
             new Input(
               hintText: 'What do people call you?',
               labelText: 'Name',
-              formField: new FormField<String>(
-                // TODO(mpcomplete): replace with person#name=
-                setter:  (String val) { person.name = val; },
-                validator: _validateName
-              )
```

Comment:
One problem is that the value of an Input (an InputValue) has a selection, but the data model is a String, so it can't be the exclusive ground truth.

Another complication is with forms that you prepare and then submit (vs those that are "live"). For example the expansion panel example used to save even if you hit "Cancel"; this patch attempts to fix that.

But yeah, I don't like that you have to have all these FormField objects everywhere.